daemon/src/taker_cfd.rs

```diff
@@ -60,6 +60,29 @@ enum RollOverState {
     None,
 }
 
+pub struct AutoRolloverParams {
+    settlement_interval: Duration,
+    /// Defines the minimum duration a CFD needs to be open for until a roll-over is triggered.
+    max_open_until_rollover: Duration,
+}
+
+impl AutoRolloverParams {
+    pub fn new(rollover_interval: Duration, settlement_interval: Duration) -> Result<Self> {
+        Ok(Self {
+            settlement_interval,
+            max_open_until_rollover: rollover_interval,
+        })
+    }
+
+    // Two conditions need to be bet for a roll over to be triggered.
+    // The cfd should not of run longer that the settlement interval.
+    // The cfd should of lived longer than the rollover interval.
+    pub fn is_cfd_due_for_rollover(&self, elapsed_since_cfd_opened: Duration) -> bool {
```

Comment:
I would find it more helpful if the complete time handling would be encapsulated in this function. 
i.e. I would pass in `cfd.expiry_timestamp()` and all the rest should happen inside this function.

The current split up is very hard to follow.